dotnet/src/SemanticKernel/Planning/FunctionFlowParser.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// Parse XML plans created by the Function Flow semantic function.
+/// </summary>
+internal static class FunctionFlowParser
+{
+    /// <summary>
+    /// The tag name used in the plan xml for the user's goal/ask.
+    /// </summary>
+    internal const string GoalTag = "goal";
+
+    /// <summary>
+    /// The tag name used in the plan xml for the solution.
+    /// </summary>
+    internal const string SolutionTag = "plan";
+
+    /// <summary>
+    /// The tag name used in the plan xml for a step that calls a skill function.
+    /// </summary>
+    internal const string FunctionTag = "function.";
+
+    /// <summary>
+    /// The attribute tag used in the plan xml for setting the context variable name to set the output of a function to.
+    /// </summary>
+    internal const string SetContextVariableTag = "setContextVariable";
+
+    /// <summary>
+    /// The attribute tag used in the plan xml for appending the output of a function to the final result for a plan.
+    /// </summary>
+    internal const string AppendToResultTag = "appendToResult";
+
+    internal static SequentialPlan ToPlanFromXml(this string xmlString, SKContext context)
```

Comment:
TODO - more testing and integration of Roger's Conditional work -- might need some help here.

Reply:
Maybe we can just have Conditional stuff be their own Plan implementation that can be added to a SequentialPlan or other.